compiler/passes/analyses/track_volume.py

```diff
@@ -0,0 +1,252 @@
+import networkx as nx
+
+from compiler.data_structures import Program
+from compiler.passes.analyses.bs_analysis import BSAnalysis
+from compiler.data_structures.ir import *
+
+
+class VolumeTracker(BSAnalysis):
+
+    variable_volume = dict();                # The dict that will store the volumes of relevent variables
+
+    violation_found = False
+
+    _program = None;

```

Comment:
These belong in `__init__()`

Reply:
If I move the variable declarations to within init() then I can't access them later on in my code.

Reply:
If they are not bound to self, then that is true; however, if you declare them, for example, as "self.variable_volume = dict()" and so on, then they are bound to the object instance.